src/main/java/it/pagopa/pagopa/apiconfig/service/CreditorInstitutionsService.java

```diff
@@ -111,14 +122,14 @@ public CreditorInstitutionStationEdit createCreditorInstitutionStation(String cr
             throw new AppException(AppError.RELATION_STATION_CONFLICT, creditorInstitutionCode, creditorInstitutionStationEdit.getStationCode());
         }
 
-        if (!paStazionePaRepository.findAllByFkPaAndSegregazione(pa.getObjId(), creditorInstitutionStationEdit.getSegregationCode()).isEmpty()) {
-            throw new AppException(HttpStatus.CONFLICT, "Bad Relation info", "SegregationCode already exists");
-        }
-        if (!paStazionePaRepository.findAllByFkPaAndProgressivo(pa.getObjId(), creditorInstitutionStationEdit.getApplicationCode()).isEmpty()) {
-            throw new AppException(HttpStatus.CONFLICT, "Bad Relation info", "ApplicationCode already exists");
-        }
+        // check uniqueness rules
+        checkSegregationPresent(creditorInstitutionStationEdit, pa);
+        checkApplicationCodePresent(creditorInstitutionStationEdit, pa);
 
         // add info into object for model mapper
+        if (creditorInstitutionStationEdit.getAuxDigit().equals(0L) || creditorInstitutionStationEdit.getAuxDigit().equals(3L)) {
```

Comment:
could you generalize this section with 158-160 lines?

Reply:
sure